### PR TITLE
Remove match command

### PR DESCRIPTION
**Pull request by ioanatia**
We decided to replace the `match` command with the `qstr` function (added in #112590)

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java [hunk 1]
```diff
@@ -256,11 +256,6 @@ public enum Cap {
          */
         COMBINE_BINARY_COMPARISONS,
 
-        /**
-         * MATCH command support
-         */
-        MATCH_COMMAND(true),
-
```
Comment (ChrisHegarty) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:263]: I think that it is safe to remove the capability. @thecoop  ??

Comment (thecoop) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:263]: Yeah, capabilities are dynamic and can be removed if they no longer apply

**Review by carlosdelest — APPROVED** [review]
LGTM ✂️

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Review by fang-xing-esql — APPROVED** [review]
LGTM, just wait for CI to be cleared, thank you!

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/release-tests

**Comment by ioanatia** [comment]
For the release-tests, the failures are related to https://github.com/elastic/elasticsearch/issues/113496 which don't have to do with the changes we have here.
We don't have any other ES|QL tests failing.
I don't see anything else in the release tests related to ES|QL - we have some other tasks for security/ML etc.
So this should be good to merge!

**Pull request merged**